launch/pipeline/service.py

```diff
@@ -0,0 +1,76 @@
+import inspect
+from typing import Any, Callable, Dict, List, Optional
+
+from launch.pipeline.deploy import Deployment
+from launch.pipeline.runtime import Runtime
+
+
+class ServiceDescription:
+    """
+    Base ServiceDescription.
+    """
+
+    def call(self, *args, **kwargs):
+        raise NotImplementedError()
+
+
+class SingleServiceDescription(ServiceDescription):
+    """
+    The description of a service.
+    """
+
+    def __init__(
+        self,
+        service: Callable,
+        runtime: Runtime,
+        deploy: Deployment,
+        init_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        super().__init__()
+        self.service = service
+        self.runtime = runtime
+        self.deploy = deploy
+        self.init_kwargs = init_kwargs
+
+        self._callable_service: Optional[Callable] = None
+
+    def _init_callable_service(self):
+        if self._callable_service:
+            return
+        if inspect.isclass(self.service):
+            kwargs = self.init_kwargs or {}
+            self._callable_service = self.service(**kwargs)
+        else:
+            assert (
+                not self.init_kwargs
+            ), "`init_kwargs` is given, but the service is not a class"
+            self._callable_service = self.service
+
+    def call(self, *args, **kwargs):
```

Comment:
Is this intended to be called by the code running in our services? Or just locally?

Reply:
Cool, from what I can tell, I think this should be compatible with the API that I provide at https://github.com/scaleapi/models/pull/2991/files#r851553522? (specifically a `make_request` function that takes in a Servable identifier (that I've been using function names for), function, and function args. Would like to check though.